irony.el

```diff
@@ -661,6 +661,17 @@ If no such file exists on the filesystem the special file '-' is
                              (format "%s\n"
                                      (combine-and-quote-strings argv)))))))
 
+(defun irony--send-request-sync (request callback &rest args)
+  "Send a request to irony-server and wait for the result.
+
+callback should set the irony-sync-result property of the irony-server process"
+  (let ((process (irony--get-server-process-create)))
+    ;; Using apply to "spread" the args list
+    (apply 'irony--send-request request callback args)
+    ;; While there are callbacks left on the queue, wait for output from server
+    (while (process-get process 'irony-callback-stack)
+      (accept-process-output process))))
```

Comment:
You need to handle some kind of local quit. Quoting `4.` from https://github.com/Sarcasm/irony-mode/issues/176#issuecomment-77773326

> when using accept-process-output, C-g should be allowed to not freeze Emacs if something goes wrong, see https://www.gnu.org/software/emacs/manual/html_node/elisp/Timers.html. FYI, a long time ago, irony used to be blocking and I was using this function to wait for server response: https://github.com/Sarcasm/irony-mode/blob/0111357aff8c30a7fa54393365e822274f0ca739/elisp/irony.el#L371

Notably the function checks whether or not the process died, I'm not sure if that's necessary but that may be.